server/py/services/api/launcher.py

```diff
@@ -239,6 +240,15 @@ def _pre_run_node_selector_enrichement(self, runtime, run):
             run.spec.node_selector = resolved_node_selectors
         return run
 
+    def _pre_run_image_pull_secret_enrichment(self, runtime, run):
+        """
+        Enrich the run object with the project's image pull secret.
+        This ensures the image pull secret is correctly set on the run
+        while maintaining the runtime's integrity from system-specific project settings.
+        """
+        run.spec.image_pull_secret = mlrun.config.config.function.spec.image_pull_secret
```

Comment:
```suggestion
        run.spec.image_pull_secret = mlrun.config.config.function.spec.image_pull_secret
```

the side effect is that if client set an explicit image_pull_secret you override it

Reply:
fixed